### PR TITLE
feat(Menu): Added MenuExtraContent

### LINKED ISSUE #349: Menu: non-selectable menu item
Add non-selectable menu item component as we already have CSS for it https://github.com/iTwin/iTwinUI/pull/312
Would be great to improve `Header` story with it:
image

### PR DESCRIPTION
**Pull request by bentleyvk**
image

Closes #349  

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] Add component features demo in Storybook (different stories)
- [x] Approve test images for new stories
- [x] Add screenshots of the key elements of the component

### CODE REVIEW

File: stories/core/DropdownMenu.stories.tsx [hunk 1]
```diff
@@ -314,3 +318,52 @@ export const WithSeparator: Story<DropdownMenuProps> = (args) => {
...
+  const [userType, setUserType] = useState('User');
+
+  const dropdownMenuItems = (close: () => void) => [
+    <MenuContent key={0}>
+      <>
+        <Text variant='leading'>Terry Rivers</Text>
+        <Text isMuted style={{ marginBottom: 8 }}>
+          terry.rivers@email.com
+        </Text>
+        <Select
```
Comment (mayank99) [inline stories/core/DropdownMenu.stories.tsx:339]: This Select gets focused using arrow keys. Seems wrong to me as it's not a list item.

File: src/core/Menu/MenuContent.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import React from 'react';
+import { useTheme } from '../utils';
+import '@itwin/itwinui-css/css/menu.css';
+
+export type MenuContentProps = {
+  children: React.ReactNode;
+};
+
+/**
+ * Component that allows to have any additional content inside `Menu`.
```
[the remark this answers is not in the document]
Comment (veekeys) [reply src/core/Menu/MenuContent.tsx:14]: It is same as css class.. Any better suggestions?

File: stories/core/DropdownMenu.stories.tsx (hunk 1 above)
Comment (veekeys) [reply stories/core/DropdownMenu.stories.tsx:339]: I think when menu opened, first focusable element should be focused (even if its in non interactive menu item)..no??

File: src/core/Menu/MenuContent.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (bentleyvk) [reply src/core/Menu/MenuContent.tsx:14]: I also don't like the name but I can't think anything better.

File: stories/core/DropdownMenu.stories.tsx [hunk 3]
```diff
@@ -20,6 +23,7 @@ import {
   SvgPlaceholder,
 } from '@itwin/itwinui-icons-react';
 import { CreeveyMeta, CreeveyStoryParams } from 'creevey';
+import { useState } from '@storybook/addons';
 
 export default {
   title: 'Core/DropdownMenu',
```
[the remark this answers is not in the document]
Comment (bentleyvk) [reply stories/core/DropdownMenu.stories.tsx:29]: Added

File: src/core/Menu/MenuContent.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ *       Sign out
+ *     </MenuItem>,
+ *   ]}
+ * </Menu>
+ */
+export const MenuContent = (props: MenuContentProps) => {
+  const { children } = props;
+  useTheme();
+  return (
+    <li className='iui-menu-content' role='presentation'>
```
[the remark this answers is not in the document]
Comment (bentleyvk) [reply src/core/Menu/MenuContent.tsx]: Added

File: stories/core/DropdownMenu.stories.tsx (hunk 1 above)
Comment (bentleyvk) [reply stories/core/DropdownMenu.stories.tsx:339]: Created an issue: https://github.com/iTwin/iTwinUI-react/issues/379

**Review by mayank99 — APPROVED** [review]

**Pull request merged**